### PR TITLE
lab: add page

### PR DESCRIPTION
**Pull request by Samox**
Main docs link:
https://github.com/zaquestion/lab/blob/master/docs/index.md

### CODE REVIEW

**Comment by pxgamer** [comment]
Hi @Samox, just wanted to check, is the correct CLI tool that this PR references?
https://github.com/lighttiger2505/lab

File: pages/osx/lab.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+> A CLI for Gitlab.
+
+- Create a merge request on origin master:
+
+`lab merge-request`
+
+- Same as above but specify remote and target branch:
+
+`lab merge-request origin develop`
```
Comment (agnivade) [inline pages/osx/lab.md]: I see this in the documentation `-b feature/feature-name -t develop`. Are the `-b` and `-t` flags not needed ?

Comment (Samox) [reply pages/osx/lab.md]: Hi Agnivade. Thanks for the feedback I made some update :)

No the flag are not needed. I forgot to mention that a [PR on lab](https://github.com/zaquestion/lab/pull/128/files) must be merged before this doc is relevant (there is a bug atm). Shall I close this PR for now and reopen later ?

**Comment by pxgamer** [comment]
Ah, so it's a Go package not Node (I've added a docs link to the PR description).

As this hasn't been merged yet, I'd suggest maybe adding some of the other available options for now, and then making a PR in future once the other PR you mentioned has been merged.

It's up to you though.

**Comment by agnivade** [comment]
The page should be in common first of all.

Secondly, you can always change the documentation to include the flags, and then later send another PR to update it.

Also,  please use tokens for variables. https://github.com/tldr-pages/tldr/blob/master/contributing-guides/style-guide.md#token-syntax

**Comment by Samox** [comment]
I updated the doc. And it is using a command that works and does not need the next [update](https://github.com/zaquestion/lab/pull/128) :)

File: pages/common/lab.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+> A CLI for Gitlab.
+
+- Create a merge request and specify remote and target branch:
+
+`lab mr create {{remote}} {{target_branch}}`
+
+- Create a pull request on a github repo:
+
+`lab pull-request`
```
Comment (zaquestion) [inline pages/common/lab.md]: Just a note, this and all the below functions are actually https://github.com/github/hub commands. `lab` can call into `hub` commands when its installed.

File: pages/common/lab.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# lab
+
+> A CLI for Gitlab.
+
+- Create a merge request and specify remote and target branch:
+
+`lab mr create {{remote}} {{target_branch}}`
+
+- Create a pull request on a github repo:
```
Comment (agnivade) [inline pages/common/lab.md]: Shouldn't it be "gitlab" repo ? And we should rename it to "merge request" in keeping parity with gitlab terminology.

File: pages/common/lab.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+`lab pull-request`
+
+- Open the repo in your browser:
+
+`lab browse`
+
+- After a fork, compare your branch with the origin branch in your browser:
```
Comment (agnivade) [inline pages/common/lab.md]: Is it compulsory to have a fork ? Can't it work when the repo is cloned ?

Comment (Samox) [reply pages/common/lab.md]: Nop, I was thinking that my use case was the most usefull. I'll update

**Comment by Samox** [comment]
I have amended the last commit following your comments.

File: pages/common/lab.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+- Create a merge request and specify remote and target branch:
+
+`lab mr create {{remote}} {{target_branch}}`
+
+- The `lab` CLI can call into `hub` commands. Create a pull request on a github repo:
+
+`lab pull-request`
+
+- Open the repo in your browser:
```
Comment (agnivade) [inline pages/common/lab.md]: in "a" browser.

File: pages/common/lab.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+`lab browse`
+
+- Compare your branch with the origin branch in your browser:
```
Comment (agnivade) [inline pages/common/lab.md]: Compare the local branch with the origin branch in a browser.

File: pages/common/lab.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+`lab compare {{username}}/{{branch}} {{target_branch}}`
```
Comment (agnivade) [inline pages/common/lab.md]: s/branch/local_branch

File: pages/common/lab.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+> A CLI for Gitlab.
+
+- Create a merge request and specify remote and target branch:
+
+`lab mr create {{remote}} {{target_branch}}`
+
+- The `lab` CLI can call into `hub` commands. Create a pull request on a github repo:
+
+`lab pull-request`
```
Comment (agnivade) [inline pages/common/lab.md]: Do we not need to specify the remote branch ?

Comment (zaquestion) [reply pages/common/lab.md]: Nah, `hub` matches from currently checked out branch and targets `origin/master` by default

Comment (agnivade) [reply pages/common/lab.md]: Ok, we should mention that then to be clear. It is not immediately apparent.

File: pages/common/lab.md (hunk 5 above)
Comment (sbrl) [reply pages/common/lab.md]: Or even `the default browser`?

**Comment by agnivade** [comment]
@Samox - Please sign the CLA so that we can get on with merging this. 

And also, slight tip - you do not need to squash commits when you update the PR. Keeping separate commits helps us to review exactly what changed in a particular commit. And while merging the PR, we anyways have a button to squash all commits.

**Comment by Samox** [comment]
Yeah thanks for the tip, I realised it was messy after the 3rd fixup... I signed the agreement. Thanks for tldr, byebye !

**Comment by Samox** [comment]
Well I have signed it and pushed the branch again but it seems it is having issues :l

**Comment by pxgamer** [comment]
Hi @Samox, it looks like the commits have been made with an email address that isn't linked to your GitHub account. This is why the CLA is showing as not-signed (at least, that's what I can gather from the CLA comment).

**Pull request closed without merging**